fuel/src/main/kotlin/com/github/kittinunf/fuel/core/requests/UploadTaskRequest.kt

```diff
@@ -14,8 +14,8 @@ class UploadTaskRequest(request: Request) : TaskRequest(request) {
 
     val BUFFER_SIZE = 1024
 
-    val CRLF = "\\r\\n"
-    val boundary = System.currentTimeMillis().toHexString()
+    val CRLF = "\r\n"
+    val boundary = request.httpHeaders["Content-Type"]!!.split("=", limit=2)[1]
```

Comment:
In the spec, it says it **SHOULD** be there

> Content-Type specifies the media type of the underlying data.
Content-Encoding may be used to indicate any additional content
codings applied to the data, usually for the purpose of data
compression, that are a property of the requested resource. There is
no default encoding.

> Any HTTP/1.1 message containing an entity-body SHOULD include a
Content-Type header field defining the media type of that body. If
and only if the media type is not given by a Content-Type field, the
recipient MAY attempt to guess the media type via inspection of its
content and/or the name extension(s) of the URI used to identify the
resource. If the media type remains unknown, the recipient SHOULD
treat it as type "application/octet-stream".

I know that it is unlikely to happen but I would stay away for `!!`. Maybe if it is not there, we fall back to the `System.currentTimeMillis()` ?